### PR TITLE
Added support for linking accounts to sub-merchants

### PR DESCRIPTION
**Pull request by thomshutt**

### CODE REVIEW

File: src/main/java/com/brightcove/castlabs/client/CastlabsClient.java [hunk 1]
```diff
@@ -222,38 +213,75 @@ public AddSubMerchantAccountResponse addSubMerchantAccount(final AddSubMerchantA
...
+            throws IOException, CastlabsException {
+
+        final String uri = this.getUrlWithTicket(this.ingestionBaseUrl + "frontend/rest/reselling/v1/reseller/" + resellerUuid + "/submerchant/linkAccount");
+        final HttpPost httpRequest = createHttpPostRequest(uri, request);
+
+        final CloseableHttpClient httpclient = HttpClients.createDefault();
+        try (final CloseableHttpResponse httpResponse = httpclient.execute(httpRequest)) {
+            final int statusCode = httpResponse.getStatusLine().getStatusCode();
+
+            if (statusCode < 200 || statusCode >= 300) {
```
Comment (stuarthicks) [inline src/main/java/com/brightcove/castlabs/client/CastlabsClient.java]: I'd be surprised if `org.apache.http` didn't have constants for these somewhere. Could we double-check?

File: src/main/java/com/brightcove/castlabs/client/CastlabsClient.java [hunk 2]
```diff
@@ -212,7 +203,7 @@ public IngestAssetsResponse ingestKeys(final IngestKeysRequest request, final St
     /**
      * Add a sub merchant account to Castlabs.
      *
-     * @param request Request parameters to pass to Castlabs
+     * @param request      Request parameters to pass to Castlabs
```
Comment (stuarthicks) [inline src/main/java/com/brightcove/castlabs/client/CastlabsClient.java]: Is your IDE adding the whitespace here?

File: src/main/java/com/brightcove/castlabs/client/CastlabsClient.java [hunk 3]
```diff
@@ -222,38 +213,75 @@ public AddSubMerchantAccountResponse addSubMerchantAccount(final AddSubMerchantA
...
+                if (responseEntity != null) {
+                    responseBody = IOUtils.toString(responseEntity.getContent());
+                }
+
+                throw new CastlabsException("Unexpected status code from Castlabs: " + statusCode + ". Response body: " + responseBody);
+            }
+        }
+    }
+
+    private HttpPost createHttpPostRequest(final String uri, final Object body) throws JsonProcessingException, UnsupportedEncodingException {
```
Comment (stuarthicks) [inline src/main/java/com/brightcove/castlabs/client/CastlabsClient.java]: I approve of this refactoring 👍

File: src/main/java/com/brightcove/castlabs/client/CastlabsClient.java (hunk 2 above)
Comment (thomshutt) [inline src/main/java/com/brightcove/castlabs/client/CastlabsClient.java]: Yep, it lines it up with the @param on the line below

**Pull request merged**